inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/service/RecommendationServiceImpl.java

```diff
@@ -373,26 +373,28 @@ public void putTrainedModel(Recommender aRecommender, Object aModel)
             trainedModels.put(aRecommender.getId(), aModel);
         }
 
-        public void removePredictions(Recommender aRecommender, Project aProject)
+        public void removePredictions(Recommender aRecommender)
         {
             // Remove incoming predictions
-            Predictions incoming = incomingPredictions.get(aProject.getId());
+            Predictions incoming = incomingPredictions.get(aRecommender.getProject().getId());
             if (incoming != null) {
                 incoming.removePredictions(aRecommender.getId());
-                incomingPredictions.put(aProject.getId(), incoming);
+                incomingPredictions.put(aRecommender.getProject().getId(), incoming);
             }
 
             // Remove active predictions
-            Predictions active = activePredictions.get(aProject.getId());
+            Predictions active = activePredictions.get(aRecommender.getProject().getId());
             if (active != null) {
                 active.removePredictions(aRecommender.getId());
-                activePredictions.put(aProject.getId(), active);
+                activePredictions.put(aRecommender.getProject().getId(), active);
             }
 
             // Remove trainedModel
             trainedModels.remove(aRecommender.getId());
 
-            // Remove from activeRecommenders map
+            // Remove from activeRecommenders map.
+            // We have to do this, otherwise training and prediction continues for the
+            // recommender when a new task is triggered.
```

Comment:
Why would we not want the training and prediction to continue? 

If a document is reset, we need to clear the model and the current predictions, but then a new training/prediction run needs to be triggered. If we remove the recommender form the active map, this doesn't happen, does it?